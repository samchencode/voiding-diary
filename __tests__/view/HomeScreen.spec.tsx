import React from 'react';
import renderer from 'react-test-renderer';
import HomeScreenFactory from '@/view/home-screen';

describe('<HomeScreen />', () => {
  const HomeScreen = HomeScreenFactory();

  it('has 1 child', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();

    // @ts-expect-error shouldnt be null
    expect(tree.children.length).toBe(1);
  });

  it('renders correctly', () => {
    const tree = renderer.create(<HomeScreen />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
