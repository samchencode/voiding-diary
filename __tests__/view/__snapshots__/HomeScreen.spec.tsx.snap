// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HomeScreen /> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <Text>
    Hello from HomeScreen!
  </Text>
</View>
`;
